upup/pkg/fi/cloudup/awstasks/iamrole.go

```diff
@@ -147,6 +148,10 @@ func (_ *IAMRole) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *IAMRole) error
 		request.AssumeRolePolicyDocument = aws.String(policy)
 		request.RoleName = e.Name
 
+		if e.PermissionsBoundary != nil {
+			request.PermissionsBoundary = e.PermissionsBoundary
```

Comment:
Up in the Find function above (~ line 74) you'll need to read the PermissionsBoundary field from the AWS SDK's Role and set it on the `awstasks.IAMRole`, similar to name and id fields on those lines.

Reply:
Fixed. Thanks for pointing that out :)

Reply:
I realized that I wasn't addressing possible PermissionsBoundary updates in the `RenderAws` function. Added that logic too.